src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/ReadyToRunHelperNode.cs

```diff
@@ -162,6 +162,24 @@ protected override DependencyList ComputeNonRelocationBasedDependencies(NodeFact
                 dependencyList.Add(factory.TypeThreadStaticsSymbol((MetadataType)_target), "ReadyToRun Thread Static Storage");
                 return dependencyList;
             }
+            else if (_id == ReadyToRunHelperId.ResolveGenericVirtualMethod)
+            {
+                MethodDesc method = _target as MethodDesc;
+                Debug.Assert(method != null && method.HasInstantiation && method.IsVirtual);
+
+                DependencyList dependencyList = new DependencyList();
+
+                // If the target method is an exact (non-canonical) instantiation, we'll need an entry for it in the exact method instantiations 
+                // hashtable, so we append the entry's dependencies to the method here.
+                var exactMethodInstantiationDependencies = ExactMethodInstantiationsNode.GetExactMethodInstantiationDependenciesForMethod(factory, method);
```

Comment:
It should have been the other way around (this was some very old code I added to my branch). We should track the method entry point here, not the hashtable dependencies.
Hashtable entries will be created from the MethodCodeNodes that we would create from here

---
In reply to: [96722057](https://github.com/dotnet/corert/pull/2521#discussion_r96722057) [](ancestors = 96722057)